### ISSUE TITLE
round-trip issue on terms with symbolic namespace segments

### ISSUE THREAD

**Issue by ceedubs** [unisonweb/unison#3362]
A term like `Nat.==.doc`, which has a symbolic segment in its namespace (the `==` segment) doesn't properly round trip. See the following transcript.

````
```ucm
.> cd MyType.==

  ☝️  The namespace .MyType.== is empty.

```
```unison
example = "example"
```

```ucm

  I found and typechecked these definitions in scratch.u. If you
  do an `add` or `update`, here's how your codebase would
  change:
  
    ⍟ These new definitions are ok to `add`:
    
      example : ##Text

```
```ucm
.MyType.==> add example

  ⍟ I've added these definitions:
  
    example : ##Text

.MyType.==> cd .

```
```ucm
.> edit MyType.==.example

  ☝️
  
  I added these definitions to the top of
  /home/cody/code/unison/scratch.u
  
    (MyType.==.example) : ##Text
    (MyType.==.example) = "example"
  
  You can edit them there, then do `update` to replace the
  definitions currently in this namespace.

.> load scratch.u

  offset=1:
  unexpected (
  expecting ability, type, or use
      1 | (MyType.==.example) : ##Text
  

```



🛑

The transcript failed due to an error in the stanza above. The error is:


  offset=1:
  unexpected (
  expecting ability, type, or use
      1 | (MyType.==.example) : ##Text
````

**Comment by aryairani** [comment]
Expected to be fixed in #3525

**Comment by aryairani** [comment]
It seems to have been fixed.
